stats/stats/src/charts/lines/new_txns_window.rs

```diff
@@ -0,0 +1,203 @@
+//! New transactions for the last N days (usually 30).
+//!
+//! Allowed to work on a non-indexed networks, as it
+//! recalculates whole N day window/range each time.
+//!
+//! Does not include last day, even as incomplete day.
+
+use crate::{
+    data_source::{
+        kinds::{
+            local_db::{
+                parameters::{
+                    update::batching::{
+                        parameters::{BatchMaxDays, ClearAllAndPassStep},
+                        BatchUpdate,
+                    },
+                    DefaultCreate, DefaultQueryVec,
+                },
+                LocalDbChartSource,
+            },
+            remote_db::{RemoteDatabaseSource, RemoteQueryBehaviour, StatementFromRange},
+        },
+        types::BlockscoutMigrations,
+        UpdateContext,
+    },
+    range::UniversalRange,
+    types::{Timespan, TimespanDuration, TimespanValue},
+    utils::day_start,
+    ChartError, ChartProperties, Named,
+};
+
+use chrono::{DateTime, NaiveDate, Utc};
+use entity::sea_orm_active_enums::ChartType;
+use sea_orm::{FromQueryResult, Statement};
+
+use super::NewTxnsStatement;
+
+pub const WINDOW: u64 = 30;
+
+fn new_txns_window_statement(
+    update_day: NaiveDate,
+    completed_migrations: &BlockscoutMigrations,
+) -> Statement {
+    // `update_day` is not included because the data would
+    // be incomplete.
+    let window =
+        day_start(&update_day.saturating_sub(TimespanDuration::from_timespan_repeats(WINDOW)))
+            ..day_start(&update_day);
+    // `t.block_timestamp != to_timestamp(0)` in this case doesn't introduce major
+    // performance drawbacks, as happened with `total_txns`, so we can
+    // reuse the statement.
+    // i.e. on sepolia testnet: 83s (w/ filter) vs 78s (w/o)
```

Comment:
but i didnt find `!= to_timestamp(0)` in NewTxnsStatement. it has comment where you said that you deleted it

Reply:
true, forgot about it since I made these PRs in parallel